sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/plan/strategy/CanaryStrategy.java

```diff
@@ -1,58 +1,196 @@
 package com.mesosphere.sdk.scheduler.plan.strategy;
 
 import com.mesosphere.sdk.scheduler.plan.Element;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import com.mesosphere.sdk.scheduler.plan.ParentElement;
+import com.mesosphere.sdk.scheduler.plan.PlanUtils;
+import com.mesosphere.sdk.scheduler.plan.Step;
 
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
 import java.util.List;
-import java.util.Optional;
-import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.stream.Collectors;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 /**
  * A CanaryStrategy blocks deployment entirely by default until human intervention through the {@link #proceed()} call.
  * After a single {@link #proceed()} call it again blocks until human intervention.  Any further calls to
  * {@link #proceed()} will indicate the strategy should continue without further intervention.
  *
- * @param <C> is the type of {@link Element}s to which the Strategy applies.
+ * Unlike other strategies, CanaryStrategy may only be applied to {@link com.mesosphere.sdk.scheduler.plan.Phase}s to
+ * manage the progress of child {@link Step}s. This is a measure to avoid user/developer error resulting from ambiguity
+ * in what should happen when a CanaryStrategy is applied to a {@link com.mesosphere.sdk.scheduler.plan.Plan}. In
+ * practice developers and users should be able to get their desired behavior by directly applying CanaryStrategy to
+ * some subset of the {@link Phases}s in their plan.
  */
-@SuppressWarnings("rawtypes")
-public class CanaryStrategy<C extends Element> extends SerialStrategy<C> {
+public class CanaryStrategy implements Strategy<Step> {
+
+    /**
+     * One proceed() to launch the first block, then a second proceed() to launch all the rest
+     */
+    private static final int DEFAULT_PROCEED_COUNT = 2;
+
     private final Logger logger = LoggerFactory.getLogger(getClass());
-    private AtomicBoolean initialized = new AtomicBoolean(false);
-    private List<Element> children;
+    private final int requiredProceeds;
+    private final Strategy<Step> postCanaryStrategy;
+
+    private List<InterruptableStep> canaryProceedSteps;
```

Comment:
Yes it's just bookeeping of the steps that need to be interrupted for the canary itself before falling back to the provided strategy. Interrupt isn't quite accurate as these steps just haven't been started yet when interrupted=true.